utils.py

```diff
@@ -0,0 +1,160 @@
+# General imports.
+import os
+import zipfile
+from glob import glob
+from shutil import copyfile
+from shutil import copytree
+from shutil import copyfileobj
+from shutil import move
+import h5py
+import numpy as np
+from skimage.measure import regionprops
+import requests
+from tqdm import tqdm
+# saving images to tif formats
+import imageio.v3 as imageio
+
+# this is the folder where all data is stored.
+data_folder = "data"
+os.makedirs(data_folder, exist_ok=True)
+
+# the data is on zenodo
+data_url = "https://zenodo.org/record/5092850/files/covid-if-groundtruth.zip?download=1"
+
+# download the data using the requests library
+
+# this function is for downloading the data.
+# you don't need to understand what's going on here.
+def download_url(url, path):
+    if os.path.isfile(path):
+        return 
+    with requests.get(url, stream=True) as r:
+        if r.status_code != 200:
+            r.raise_for_status()
+            raise RuntimeError(f"Request to {url} returned status code {r.status_code}")
+        file_size = int(r.headers.get("Content-Length", 0))
+        desc = f"Download {url} to {path}"
+        if file_size == 0:
+            desc += " (unknown file size)"
+        with tqdm.wrapattr(r.raw, "read", total=file_size, desc=desc) as r_raw, open(path, "wb") as f:
+            copyfileobj(r_raw, f)
+
+
+# run the download function to donwnload the data to the file 'data.zip' in the data folder
+download_url(data_url, os.path.join(data_folder, "data.zip"))
```

Comment:
Python script files should not contain any call to functions or variable definitions on the level. Otherwise these functions will be called every time we import some function from this file.

Reply:
In order to execute some functions in a script, but still enable importing from it you can use the following construction at the end of the file:
```python
if __name__ == "__main__":
   ...
```
The code in the if will only be executed if the script is directly run with a python executor, i.e. `python utils.py`, but not if we import from this file.

Reply:
In this case I think it's best if we add a new high-level function that takes the `data_folder` as input and then downloads, extracts and rewrites the data.